src/NServiceBus.AcceptanceTesting/Support/WhenDefinition.cs

```diff
@@ -5,14 +5,30 @@ namespace NServiceBus.AcceptanceTesting.Support
 
     public class WhenDefinition<TContext> : IWhenDefinition where TContext : ScenarioContext
     {
+        // remove in v8
```

Comment:
are you suggesting to change all `When` APIs to use async conditions in the future or were you thinking about changing these functions: https://github.com/Particular/NServiceBus/pull/5213/files#diff-632ce5abc1e2165d5b91f08ec888ef5fR34 to be changed to map the non-async overload to apply the transformation to async?

For the later, is there any major reason to not do that right away?

Reply:
I was thinking about the ctors of the When that they should only be using the async ones and the public API can still bridge the sync version to the async one. Clearer?